Detectors/Raw/src/RawFileWriter.cxx

```diff
@@ -726,3 +727,11 @@ void RawFileWriter::DetLazinessCheck::completeLinks(RawFileWriter* wr, const IR&
   }
   clear();
 }
+
+void o2::raw::assertOutputDirectory(std::string_view outDirName)
+{
+  std::filesystem::create_directories(outDirName);
```

Comment:
I think the logic here should be:
```
if (!exists) {
#ifdef MACOS
  create();
  if (!exists) error;
#else
  if (!create()) error;
#endif
}
```
with some comment that this is broken on mac OS currently and that the workaround should be reverted once it works. I like merging all the checks into this one function.

Reply:
Well, yes, that's another option. But I believe the check should be done on the compiler, not the platform, as I think the issue is more a Clang vs GCC one (I understand we're only using Clang on Mac, but the intent of the check would be clearer that way)

Reply:
Yes, sure, my point was just we should document why it is done like this, together with a comment that it should be cleaned up. Is there actually any reference that this is broken on Mac / clang? Clang is usually well-tested, so I would be surprised if `create_directories` would be misbehaving.

Reply:
Would have to dig a reference if there is one, but can be seen in e.g. https://godbolt.org/z/qcE5fMq36

Reply:
OK, then it is most likely a bug in clang?
Can you open a bug report? The clang developers are usually quite responsive.

Reply:
And still, could we change it to have the exists check first, and then add a comment with a todo that this is a workaround due to a clang bug?

Reply:
I've just added a commit (to be squashed on merge) in that respect : would that be ok for you ?